### DISCUSSION TITLE
Qual a relação entre tipos de relacionamentos entre tabelas (1-1; 1-M; M-M) e tipos de Joins?

### DISCUSSION THREAD

**Discussion by depaula36** [balta-io/forum#553] [Tira Dúvidas] [answered]
Professor, qual a relação entre tipos de relacionamentos entre tabelas (1-1; 1-M; M-M) e tipos de Joins? Fico sempre um pouco confuso ao abordar esse tema. A dúvida me retornou ao assistir as aulas referentes ao Dapper. 
A lógica seria assim (abaixo)?
1 - O tipo de Join é uma demanda independente (ou seja, pode ser solicitado qualquer um dependendo dos dados que vou querer apresentar).
2 - Para que os dados possam ser apresentados corretamente devo concluir as relações existentes entre as tabelas (1-1; 1-M; M-M) e defini-las conforme as técnicas expostas nas aulas para o Dapper.

Dúvida adicional : Tanto mundo SQL puro quando do Dapper, sou obrigado a definir as relações existentes entre as tabelas (1-1; 1-M; M-M) apenas quando quero realizar Joins? (ou seja, desejo colunas entre tabelas diferentes mas que tenham relação entre si por alguma coluna em comum). Se não for realizar Joins, estabelecer relação entre tabelas é opcional?

Desde já agradeço à atenção e parabenizo pela qualidade das aulas.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @depaula36, como vai?

Essa parte de relações é a aplicação da teoria dos conjuntos da matemática:
https://www.todamateria.com.br/teoria-dos-conjuntos/

É que normalmente nós chamamos de "tabelas" o que na verdade são relações, então se você observar o material acima, vai ver que uma tabela nada mais é do que uma relação que pode ser disjunta, unificada ou intersecta.

Ai definimos estas relações como 1:1, 1:n e NxN (Ou simplesmente não existindo). Daí vem os nomes "Join", "Inner Join", "Outer Join".

Veja se ficou mais claro!

**Reply by depaula36** [reply]
Olá, professor. Tudo bem por aqui. Ficou sim, obrigado. Grande abraço.
